core/amber/src/main/scala/edu/uci/ics/texera/web/service/WorkflowService.scala

```diff
@@ -153,16 +165,18 @@ class WorkflowService(
         )
       )
     }
-    new WorkflowContext(jobID, uidOpt, wId)
+    val vId = getLatestVersion(UInteger.valueOf(wId)).intValue()
+    new WorkflowContext(jobID, uidOpt, vId, wId)
   }
 
   def initJobService(req: WorkflowExecuteRequest, uidOpt: Option[UInteger]): Unit = {
     if (jobService.getValue != null) {
       //unsubscribe all
       jobService.getValue.unsubscribeAll()
     }
+    workflowContxt = createWorkflowContext(req)
```

Comment:
this step is refactored in a new PR #1592

Reply:
I suggest you make this PR be merged into `execution_master`

Reply:
since workflow context now has the vid, let's remove calculation of getting latestVID in the `insertExecution` function